azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -79,6 +85,40 @@ def __init__(self, mqtt_pipeline, http_pipeline):
         self._mqtt_pipeline = mqtt_pipeline
         self._http_pipeline = http_pipeline
 
+        self._inbox_manager = None  # this will be overriden in child class
+        self._receive_type = RECEIVE_TYPE_NONE_SET
+
+        # Generic Client handlers
+        self._on_method_request_received = None
+        self._on_twin_desired_properties_patch_received = None
+        self._on_message_received = None
+
+    def _validate_receive_api_invoke(self):
```

Comment:
I hate it when people open this can of worms, especially since a problem is extremely unlikely, but this function isn't thread safe and there's nothing protecting it on the calling side.  Same for the other _validate function.

Reply:
Good catch

---
In reply to: [454403531](https://github.com/Azure/azure-iot-sdk-python/pull/620#discussion_r454403531) [](ancestors = 454403531)